src/frontend/test_runner/tests/testdata/tpch.yaml

```diff
@@ -365,34 +365,32 @@
-                BatchHashJoin { type: LeftSemi, predicate: $0 = $2 }
-                  BatchProject { exprs: [$0, $2], expr_alias: [ ,  ] }
-                    BatchExchange { order: [], dist: HashShard([0]) }
-                      BatchFilter { predicate: ($1 >= '1997-07-01':Varchar::Date) AND ($1 < ('1997-07-01':Varchar::Date + '3 mons 00:00:00':Interval)) }
-                        BatchScan { table: orders, columns: [o_orderkey, o_orderdate, o_orderpriority] }
-                  BatchProject { exprs: [$0], expr_alias: [ ] }
-                    BatchExchange { order: [], dist: HashShard([0]) }
-                      BatchFilter { predicate: ($1 < $2) }
-                        BatchScan { table: lineitem, columns: [l_orderkey, l_commitdate, l_receiptdate] }
+        BatchHashAgg { group_keys: [$0], aggs: [count] }
+          BatchProject { exprs: [$1] }
+            BatchExchange { order: [], dist: HashShard([1]) }
+              BatchHashJoin { type: LeftSemi, predicate: $0 = $2 }
+                BatchProject { exprs: [$0, $2] }
+                  BatchExchange { order: [], dist: HashShard([0]) }
+                    BatchFilter { predicate: ($1 >= '1997-07-01':Varchar::Date) AND ($1 < ('1997-07-01':Varchar::Date + '3 mons 00:00:00':Interval)) }
+                      BatchScan { table: orders, columns: [o_orderkey, o_orderdate, o_orderpriority] }
+                BatchProject { exprs: [$0] }
+                  BatchExchange { order: [], dist: HashShard([0]) }
+                    BatchFilter { predicate: ($1 < $2) }
+                      BatchScan { table: lineitem, columns: [l_orderkey, l_commitdate, l_receiptdate] }
   stream_plan: |
-    StreamMaterialize { columns: [o_orderpriority, order_count], pk_columns: [o_orderpriority] }
-      StreamProject { exprs: [$0, $2], expr_alias: [o_orderpriority, order_count] }
-        StreamHashAgg { group_keys: [$0], aggs: [count, count] }
-          StreamProject { exprs: [$1, $2], expr_alias: [ ,  ] }
-            StreamExchange { dist: HashShard([1]) }
-              StreamHashJoin { type: LeftSemi, predicate: $0 = $3 }
-                StreamProject { exprs: [$0, $2, $3], expr_alias: [ ,  ,  ] }
-                  StreamExchange { dist: HashShard([0]) }
-                    StreamFilter { predicate: ($1 >= '1997-07-01':Varchar::Date) AND ($1 < ('1997-07-01':Varchar::Date + '3 mons 00:00:00':Interval)) }
-                      StreamTableScan { table: orders, columns: [o_orderkey, o_orderdate, o_orderpriority, _row_id#0], pk_indices: [3] }
-                StreamProject { exprs: [$0, $3], expr_alias: [ ,  ] }
-                  StreamExchange { dist: HashShard([0]) }
-                    StreamFilter { predicate: ($1 < $2) }
-                      StreamTableScan { table: lineitem, columns: [l_orderkey, l_commitdate, l_receiptdate, _row_id#0], pk_indices: [3] }
+    StreamMaterialize { columns: [o_orderpriority, agg#0(hidden), order_count], pk_columns: [o_orderpriority] }
```

Comment:
The old plan discards `agg#0`, which is the extra `count_star` added for stream agg. The new plan preserves it as a hidden column.

We can discuss which is more proper after this PR. The correctness holds for both plans.

Reply:
I think we should discard it and do that in the future. The agg count columns look like a special case in materialize operator. it is the only one not in `required_cols` but does not need to be materialized.